### PR TITLE
Fix view_complex_as_float for empty tensors

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37645 Fix view_complex_as_float for empty tensors**

### CODE REVIEW

File: aten/src/ATen/native/ComplexHelper.h [hunk 1]
```diff
@@ -13,28 +13,43 @@ inline std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
...
+      detail::computeStorageSize(sizes, strides),
+      DataPtr(nullptr, options.device()),
+      /*allocator=*/nullptr,
+      /*resizable=*/false);
+  return at::empty({0}, options).set_(storage, 0, sizes, strides);
+}
+
 // expects as input a complex tensor and returns back a float tensor
 // containing the complex values in the last two dimensions
 inline Tensor view_complex_as_float(const Tensor& self) {
```
Comment (gchanan) [inline aten/src/ATen/native/ComplexHelper.h:35]: this name is confusing -- it's not returning a "float" it's returning a "real" right?  I.e. the dimensions of the complex viewed as a their corresponding real components.

**Review by gchanan — CHANGES_REQUESTED** [review]
this doesn't appear to address the concern raised in https://github.com/pytorch/pytorch/pull/36415.  By the way that PR has the same name as this; please don't do that as it's confusing.

Also, this doesn't address the concern in the original PR which was:
"This fix doesn't look correct to me. I'm not 100% sure if this will cause problems, but I'd expect this could cause issues if you try to say, resize a tensor but it doesn't share the storage.

What do things like as_strided do in this case? I would guess that there are smarts around just not throwing an error, and we'd probably want to do the same thing in from_blob."

In other words, I don't think you should fix this in complex-related code.  It should be fixed in from_blob or similar code.  Thus the request to figure out how things like "as_strided" avoid this.  I don't see any analysis of that in the original PR or this PR.
